extra_geom/detectors.py

```diff
@@ -1682,19 +1682,24 @@ def from_origin(cls, origin=(0, 0), asic_gap=None, unit=None):
         if unit is None:
             unit = cls.pixel_size
         if asic_gap is None:
-            asic_gap = cls.asic_gap
+            asic_gap_h = asic_gap_v = cls.asic_gap
+        elif isinstance(asic_gap, Collection):
```

Comment:
If we're going to be all precise about types here, I think we need a `Sequence` here. `Collection` covers things like sets and dicts as well.